File: frontend/libs/model-list/src/lib/classes/properties/CommonAspectRatio.ts

/// Common aspect ratio Tauri API
/// Not all models have each of these aspect ratios, 
/// but Tauri will interpolate if a mistake is sent.
export enum CommonAspectRatio {
  // "Auto"
  // The idea here is that in image editing (not text-to-image), the model will use the source image.
  // This is the semantics used by Nano Banana Pro image editing mode (but not text-to-image).
  Auto = "auto",

  // Square
  Square = "square",

  // Wide
  WideFiveByFour = "wide_five_by_four",
  WideFourByThree = "wide_four_by_three",
  WideThreeByTwo = "wide_three_by_two",
  WideSixteenByNine = "wide_sixteen_by_nine",
  WideTwentyOneByNine = "wide_twenty_one_by_nine",

  // Tall
  TallFourByFive = "tall_four_by_five",
  TallThreeByFour = "tall_three_by_four",
  TallTwoByThree = "tall_two_by_three",
  TallNineBySixteen = "tall_nine_by_sixteen",
  TallNineByTwentyOne = "tall_nine_by_twenty_one",

  // Imprecise semantic values (that probably are mapped to other meanings)
  // Not appropriate for all models to use these values.
  Wide = "wide",
  Tall = "tall",
}

File: crates/desktop/artcraft/src/core/api_adapters/aspect_ratio/common_aspect_ratio.rs
use serde_derive::{Deserialize, Serialize};

/// This is a comprehensive list of common aspect ratios.
/// Not every model will support every aspect ratio.
/// In the case a model doesn't support the aspect ratio, pick the nearest option.
#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum CommonAspectRatio {
  // Auto (eg. for image editing to use the source; used by Nano Banana Pro edit image, but not text-to-image)
  Auto,

  // Square
  Square,

  // Wide
  WideThreeByTwo,
  WideFourByThree,
  WideFiveByFour,
  WideSixteenByNine,
  WideTwentyOneByNine,

  // Tall
  TallTwoByThree,
  TallThreeByFour,
  TallFourByFive,
  TallNineBySixteen,
  TallNineByTwentyOne,

  // Imprecise semantic values that we probably remap to other meanings
  // on a model-by-model basis.
  Wide,
  Tall,

  // Auto values that bake in resolution
  // These are from the Seedream models
  Auto2k,
  Auto4k,

  // Defined aspect ratios that bake in resolution
  // These are from the Seedream models
  SquareHd,
}
